### PR TITLE
Create Chinese_zh

### PR DESCRIPTION
**Pull request by zestybaby**
Credits to @Diprog for original Russian translation as a template and @suzumiyasmith for previous Simplified Chinese translation.

### CODE REVIEW

**Comment by zestybaby** [comment]
Noto by Google and Source Han by Adobe are both great open source fonts which supports CJK characters. In my test, I used Source Han and find it has a similar style as DejaVu.

**Comment by 1bardesign** [comment]
Noto is an incredibly large font, much larger than the entire game package!
Source Han may be acceptable, we can probably only "afford" one font weight though.

It looks like they are available from [here](https://github.com/Pal3love/Source-Han-TrueType)

**Comment by 1bardesign** [comment]
I imagine the version we want for Chinese support is CN in this case, not any of the others? We do not currently have Japanese or Korean localisations.

If it gets to the point where we would be better off using noto for size, we can always switch, but having separate ttfs seems to help a lot. I am not sure our font system would cope with a font as large as noto at this point 😅

**Comment by zestybaby** [comment]
Can confirm it's CN in this case. In fact I tested it with https://github.com/be5invis/source-han-sans-ttf/ but it's outdated comparing with the build you linked.

File: Locales/Chinese_zh [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+  "{ITEM} limit reached.": "{ITEM} 已达上限",
+  "Requires a Worker from the Hall to work here.": "需要从大厅派遣工人",
+  "Units": "单位",
+  "Another worker has been hired!": "新增雇佣工人!",
+  "Supplies will drop at your halls.": "补给会空投至大厅.",
+  "Buy a worker for {MIGRANT_COST} Gold": "以 {MIGRANT_COST} 黄金购买工人",
+  "Buy worker: Requires {MIGRANT_COST} Gold": "购买工人: 需要 {MIGRANT_COST} 黄金",
+  "Travel": "穿过隧道",
+  "Travel (requires Transport Tunnels)": "穿过隧道(需要出口)",
+  "Pick tunnel to travel": "挖掘隧道以便通行",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: should be "选择前往的隧道”

File: Locales/Chinese_zh [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+  "Return to base": "返回基地",
+  "Travel right": "向右移动",
+  "Travel left": "向左移动",
+  "You are here": "已在此处",
+  "Click the items you vote to research!": "点击选择研发项目",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: could be "点击投票选择研发项目”

File: Locales/Chinese_zh [hunk 3]
```diff
@@ -0,0 +1,890 @@
...
+  "Yellow paths show what will be researched when the match starts.": "黄色标示即将研发项目",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: could be "黄色线路标示即将研发的项目"

File: Locales/Chinese_zh [hunk 4]
```diff
@@ -0,0 +1,890 @@
...
+  "BUILD WORKSHOPS\n           $BUILD$": "建造工坊\n           $BUILD$",
+  "BUILD DEFENSES\n           $BUILD$": "建造防御工事\n           $BUILD$",
+  "CAPTURE THE HALL!\n           $DEFEND_THIS$": "占领大厅!\n           $DEFEND_THIS$",
+  "$ATTACK_LEFT$ATTACK!": "$ATTACK_LEFT$进攻!",
+  "ATTACK!$ATTACK_RIGHT$": "进攻!$ATTACK_RIGHT$",
+  "ATTACK!\n           $ATTACK_THIS$": "进攻!\n           $ATTACK_THIS$",
+  "Bombs for Knight only.": "骑士专用的炸弹.",
+  "Arrows for Archer and mounted bow.": "弓手与弩使用的箭.",
+  "Highly explosive powder keg for Knight only.": "骑士专用的桶装炸药.",
+  "A stone throwing, ridable siege engine, requiring a crew of two.": "可驾驶投石的攻城器械，需要一至两人操作.",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: should be ""可驾驶的投石机，需要一至两人操作."

File: Locales/Chinese_zh [hunk 5]
```diff
@@ -0,0 +1,890 @@
...
+  "A bolt-firing pinpoint accurate siege engine, requiring a crew of two. Allows respawn and class change.": "可射箭的精准打击攻城器械 需要两人操作 可以作为重生点与更换职业.",
+  "Piercing bolts for ballista.": "弩车使用的箭.",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: "弩车使用的弩箭."

File: Locales/Chinese_zh [hunk 6]
```diff
@@ -0,0 +1,890 @@
...
+  "For Ballista\nTurns its piercing bolts into a shaped explosive charge.": "弩车专用\n将它的箭升级为爆炸箭.",
```
Comment (Aristalia) [inline Locales/Chinese_zh]: "弩车专用\n将它的箭升级为爆炸弩箭."

**Comment by 1bardesign** [comment]
I went with 2 font weights of Source Han as they seem to compress pretty well (~13mb zipped for 2 >10mb fonts) and we wont have to patch often.

Please review in the test build which has the required font support, not the public build (until next release)

@Aristalia if you would like to apply those translations to the new file, please feel free, I dont think @zestybaby has a problem with it

**Comment by zestybaby** [comment]
I've edited to those suggested translations with new commit. Appreciate the suggestions, they are all great.

**Comment by 1bardesign** [comment]
Oh fantastic, sorry for all the pings then folks! 😅

File: Locales/Chinese_zh.json [hunk 7]
```diff
@@ -0,0 +1,890 @@
...
+  "Knights can use their swords to destroy wooden blocks and doors, but not stone": "骑士可以用他们的剑来摧毁木块和木门, 但是石头则不行.",
+  "Doors and siege weapons are vulnerable to explosives.": "门和围城武器都不能应对爆炸物的攻势.",
+  "Bison can be tamed with burgers.": "你可以用汉堡来驯服野牛, 没想到吧.",
+  "Resources can be resupplied by changing classes in workshops, not just at the tent!": "你也可以在工坊里更换职业来补给资源, 不一定只能在帐篷里这么做!",
+  "{COINS_QUANTITY} $COIN$ required\n": "需要 {COINS_QUANTITY} $COIN$",
+  "Migrant": "移民",
+  "Magic Orb": "魔石",
+  "Sedgwick": "Sedgewick",
+  "Princess Geti": "Geti 公主",
+  "Building in Construction": "尚在建筑",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "建筑施工中”

File: Locales/Chinese_zh.json [hunk 8]
```diff
@@ -0,0 +1,890 @@
...
+  "Archer Shop": "弓手商店",
+  "Boat Shop": "船舶商店",
+  "Builder Shop": "建筑师商店",
+  "Knight Shop": "骑士商店",
+  "Quarters": "食品店",
+  "Storage Cache": "仓储",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "储物仓库”

File: Locales/Chinese_zh.json [hunk 9]
```diff
@@ -0,0 +1,890 @@
...
+  "Team Storage": "团队储备",
+  "Tech Scrolls": "科技卷轴",
+  "Storage": "仓储",
+  "Builder": "建筑师",
+  "Knight": "骑士",
+  "Archer": "射手",
+  "Buy": "买",
+  "Construct": "建筑",
+  "Shop": "商店",
+  "Store": "商店",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: I'm not sure if this is the same as above or “储存”?

File: Locales/Chinese_zh.json [hunk 10]
```diff
@@ -0,0 +1,890 @@
...
+  "Travel Tunnel": "快速通行隧道",
+  "Transport Tunnel": "交通隧道",
+  "Boulder": "巨石",
+  "Bucket": "桶",
+  "Filled Bucket": "填满的桶",
+  "Crate": "箱",
+  "Bomb": "炸弹",
+  "Keg": "桶装炸药",
+  "Mine": "地雷",
+  "Satchel": "肩包",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: should be something closer to "炸药包”; before satchels were removed early in KAG beta they were molotov/bomb combinations for knights similar to fire arrows for archers.

File: Locales/Chinese_zh.json [hunk 11]
```diff
@@ -0,0 +1,890 @@
...
+  "Kill the Necromancer": "杀死死灵法师!",
+  "Parkour": "跑酷",
+  "Save the Princess!": "拯救公主!",
+  "Save the Princess": "拯救公主",
+  "Take the Hall": "占领大厅",
+  "All Gamemodes": "所有游戏模式",
+  "Challenges": "挑战",
+  "Capture the Flag": "夺旗模式",
+  "CTF": "夺旗模式",
+  "Team Deathmatch": "团队死斗",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "团队死斗模式” in consistency with "夺旗模式” above

File: Locales/Chinese_zh.json [hunk 12]
```diff
@@ -0,0 +1,890 @@
...
+  "Take the Halls": "占领大厅",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "占领大厅模式” in consistency with "夺旗模式” above

File: Locales/Chinese_zh.json [hunk 13]
```diff
@@ -0,0 +1,890 @@
...
+  "TTH": "占领大厅",
+  "Challenge": "挑战",
+  "Build whatever your heart desires. Respect other folk's creations!": "随心所欲地建造吧. 但也请尊重其他玩家的创作!",
+  "IP:Port": "IP:端口号",
+  "Password:": "密码:",
+  "Heads Pack Available On Steam!": "头像包现已登陆Steam!",
+  "Get exclusive cosmetic heads to increase your customisation options, and stand out on the battlefield!\n\nCurrently only available through steam as a DLC. If there's enough interest we can investigate non-steam options.\n\nProceeds go to continued development of King Arthur's Gold.": "获得独家装饰性的头像来增加你的自定选项, 并在战场上更加出众!\n\n目前仅在Steam上作为DLC可购买. 如果有足够多的人产生兴趣的话我们将会研究非Steam的购买方式.\n\n收入将用于继续开发亚瑟王的金子游戏.",
+  "Find Out More!": "找到更多!",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "发现更多！”

File: Locales/Chinese_zh.json [hunk 14]
```diff
@@ -0,0 +1,890 @@
...
+  "PLAY": "开始游戏",
+  "Solo": "单人模式",
+  "Multiplayer": "多人模式",
+  "Settings": "设置",
+  "How to video": "教程视频",
+  "Forum": "论坛",
+  "News": "新闻",
+  "Discord": "Discord",
+  "IRC": "IRC",
+  "Wiki": "维基",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: should be "百科” or “KAG百科”, “维基” is the wiki in wikipedia lol

File: Locales/Chinese_zh.json [hunk 15]
```diff
@@ -0,0 +1,890 @@
...
+  "E-mail:": "电子邮箱",
+  "Submit": "提交",
+  "Simple Join": "快速加入",
+  "Browse servers": "查看服务器列表",
+  "Connect to...": "连接到...",
+  "New Game From THD:": "来自THD的新游戏:",
+  "Pick Game Size:": "选择游戏大小:",
+  "<8 Players": "少于8人",
+  "8-20 Players": "8至20人",
+  ">20 Players": "多于20 人",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: "多于20人", think you accidentally added a space

File: Locales/Chinese_zh.json [hunk 16]
```diff
@@ -0,0 +1,890 @@
...
+  "Choose Game Mode:": "选择游戏模式:",
+  "Sandbox": "沙盒",
+  "Co-op Challenge": "合作挑战",
+  "Custom Content": "自定义内容",
+  "Play!": "开始游戏!",
+  "Controls": "控制",
+  "Video": "视频",
+  "Sound": "声音",
+  "Misc": "杂项",
+  "Clantag:": "战队名:",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "工会名：”

File: Locales/Chinese_zh.json [hunk 17]
```diff
@@ -0,0 +1,890 @@
...
+  "Gender:": "性别:",
+  "Male": "男性",
+  "Female": "女性",
+  "BUY {HEADS_PACK}": "购买 {HEADS_PACK}",
+  "Standard": "标准",
+  "Flags of The World": "世界旗",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: maybe "世界旗帜” or "世界之旗”? Seems slightly awkward to me

File: Locales/Chinese_zh.json [hunk 18]
```diff
@@ -0,0 +1,890 @@
...
+  "Head packs:": "头像包:",
+  "Head:": "头部:",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: should be "头像” in reference to “头像包”

File: Locales/Chinese_zh.json [hunk 19]
```diff
@@ -0,0 +1,890 @@
...
+  "Faster graphics": "快速图像",
+  "Uncap framerate": "无上限FPS",
+  "VSync - Requires Restart": "垂直同步 - 需要重启生效",
+  "Game music": "游戏音乐",
+  "Sound Volume:": "音效音量:",
+  "Music Volume:": "音乐音量:",
+  "Volume:": "音量:",
+  "Menu music": "菜单音乐",
+  "Swap audio channels": "交换左右声道",
+  "Language (EXPERIMENTAL):": "语言 (实验支持):",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "语言（测试中）：”

File: Locales/Chinese_zh.json [hunk 20]
```diff
@@ -0,0 +1,890 @@
...
+  "Fixed camera": "固定视角",
+  "Kids safe": "适合儿童",
+  "Chat Filter": "聊天消息过滤",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "聊天消息过滤器”

File: Locales/Chinese_zh.json [hunk 21]
```diff
@@ -0,0 +1,890 @@
...
+  "VSync - Requires Restart": "垂直同步 - 需要重启生效",
+  "Game music": "游戏音乐",
+  "Sound Volume:": "音效音量:",
+  "Music Volume:": "音乐音量:",
+  "Volume:": "音量:",
+  "Menu music": "菜单音乐",
+  "Swap audio channels": "交换左右声道",
+  "Language (EXPERIMENTAL):": "语言 (实验支持):",
+  "Fixed camera": "固定视角",
+  "Kids safe": "适合儿童",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "儿童模式”

File: Locales/Chinese_zh.json [hunk 22]
```diff
@@ -0,0 +1,890 @@
...
+  "Chat Filter": "聊天消息过滤",
+  "Show Team-Mates Names": "显示队伍成员的名字",
+  "Show Minimap": "显示小地图",
+  "Show Chat Bubbles": "显示聊天气泡",
+  "Snap Camera To Pixel": "拉近视角至像素",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: this means that the camera always has its borders on the edge of pixels, but I'm not sure how to translate that

File: Locales/Chinese_zh.json [hunk 23]
```diff
@@ -0,0 +1,890 @@
...
+  "KAG World": "亚瑟王的金子世界",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: I've always personally translated KAG to "亚瑟王的金“, but no significant difference really

File: Locales/Chinese_zh.json [hunk 24]
```diff
@@ -0,0 +1,890 @@
...
+  "{SERVERS} servers with {PLAYERS} players": "\"{SERVERS} 个服务器以及 {PLAYERS} 个玩家\"",
+  "Play": "玩",
+  "Spectate": "观战",
+  "OK": "确认",
+  "Beginner": "初学者",
+  "Player Count": "玩家数量",
+  "Map Size": "地图大小",
+  "Game Mode": "游戏模式",
+  "Server Name": "服务器名称",
+  "Favorites Only": "仅限最爱服务器",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "只显示收藏的服务器”

File: Locales/Chinese_zh.json [hunk 25]
```diff
@@ -0,0 +1,890 @@
...
+  "Ban - 1 hour": "封禁 - 1小时",
+  "Ban - 1 day": "封禁 - 1天",
+  "Ban - Permanent": "封禁 - 永久",
+  "Mute": "禁言",
+  "Mute - 10 mins": "禁言 - 10分钟",
+  "Mute - 1 hour": "禁言 - 1小时",
+  "Mute - 1 day": "禁言 - 1天",
+  "Mute - Permanent": "禁言 - 永久",
+  "Unmute": "取消禁言",
+  "Freeze": "静止",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be "冻结”

File: Locales/Chinese_zh.json [hunk 26]
```diff
@@ -0,0 +1,890 @@
...
+  "Unfreeze": "取消静止",
```
Comment (Aristalia) [inline Locales/Chinese_zh.json]: could be “取消冻结”

**Comment by zestybaby** [comment]
@Aristalia Appreciate all your suggestions but this translation has already been merged to the original repo, that means you need to open your on fork to work with, which also will give proper credits in the code base.

**Pull request merged**